src/main/java/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.java

```diff
@@ -252,7 +253,32 @@
      * @see #isGetHostStringPossible()
      */
     private final boolean getHostStringPossible = isGetHostStringPossible();
-    
+
+    /**
+     * The first set destination is random.
+     *
+     * Defaults to disable. The first destination is considered the primary destination.
+     */
+    private boolean randomDestination;
+
+    /**
+     * Object to create the first random index.
+     */
+    private Random firstRandomIndexGenerator = new Random();
+
+    /**
+     * If the size is exceeded, round-robin occurs.
+     * Destination cycles by round-robin.
+     *
+     * When null (the default), disabled round-robin.
+     */
+    private Integer roundRobinSize;
```

Comment:
As you know, if you do round robin, a reconnection occurs.
When reconnecting each time, a big cost was introduced, and the "roundRobinSize" was introduced as a concept of buffer.